Userland/Libraries/LibLocate/Crawler.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2021, Aron Lander <aron@aronlander.se>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <LibLocate/Crawler.h>
+
+namespace Locate {
+
+Crawler::Crawler(String path)
+{
+    m_dir_queue.enqueue(make<Locate::DirectoryInfo>(Locate::DirectoryInfo {
+        .path = path,
+        .db_id = m_id_counter,
+        .parent_db_id = 0,
+        .children = SinglyLinkedListWithCount<NonnullOwnPtr<Locate::ChildInfo>>() }));
+    m_id_counter++;
+}
+
+NonnullOwnPtr<Locate::DirectoryInfo> Crawler::index_next_directory()
+{
+    auto cur_dir_info = m_dir_queue.dequeue();
+    String cur_path = cur_dir_info->path.characters();
+    Core::DirIterator dir_iterator(cur_path, Core::DirIterator::SkipParentAndBaseDir);
+
+    while (dir_iterator.has_next()) {
+        NonnullOwnPtr<Locate::ChildInfo> child = make<Locate::ChildInfo>();
+        child->name = dir_iterator.next_path();
+
+        String fullpath;
+        if (cur_path == "/")
+            fullpath = String(String::formatted("{}{}", cur_path.characters(), child->name.characters()));
+        else
+            fullpath = String(String::formatted("{}/{}", cur_path.characters(), child->name.characters()));
```

Comment:
No need for `.characters()` when using `{}` formatting.

```suggestion
            fullpath = String(String::formatted("{}{}", cur_path, child->name));
        else
            fullpath = String(String::formatted("{}/{}", cur_path, child->name));
```